olive/passes/onnx/mnb_to_qdq.py

```diff
@@ -62,257 +64,124 @@ def _run_for_config(
-            # DQ default zp is 0 but MatMulNBits is 8, so we need to add a zero tensor with all 8s
-            # no need to add for int4 if add_zero_point is False
-            if len(dq_inputs) == 2 and (config["add_zero_point"] or not config["use_int4"]):
-                zp_name = f"{dq_name}.qzeros"
-                zp_shape = (
-                    [N] if is_per_axis else ([N, num_k_blocks] if config["use_transpose_op"] else [num_k_blocks, N])
-                )
-                zp_tensor = onnx.helper.make_tensor(
-                    zp_name,
-                    int_elem_type,
-                    zp_shape,
-                    # no zp in matmulnbits is equivalent to 8 uint4 and 0 int4 in DQ
-                    self._pack_on_flat(np.zeros(N * num_k_blocks, dtype=int_np_dtype) + 8 - default_mnb_zp).tobytes(),
-                    raw=True,
-                )
-                dag.add_initializer(zp_tensor, graph_idx)
-                dq_inputs.append(zp_name)
-
-            # onnx dtype for the float tensors (scale, dequantized weight, matmul inputs+outputs)
-            float_elem_type = onnx.helper.np_dtype_to_tensor_dtype(dag.get_initializer_np_array(node_inputs[2]).dtype)
-
-            # new nodes and value infos to add to the graph
-            # ensure that the node names and output names are unique
-            # will add the new nodes, make consumers use the new output and remove the node
-            # if output is a model output, rename it back to the original name
-            new_nodes = []
-            new_value_infos = []
-
-            # DequantizeLinear
-            dq_name = self._get_new_node_name(dag, node_name, "DequantizeLinear")
-            dq_output = f"{dq_name}/output_0"
-            new_nodes.append(
-                onnx.helper.make_node(
-                    "DequantizeLinear",
-                    dq_inputs,
-                    [dq_output],
-                    name=dq_name,
-                    block_size=None if is_per_axis else block_size,
-                    # for some reason block_wise and per-axis appear to use swapped axis
-                    # flip the axis if it is per-axis
-                    axis=(1 if config["use_transpose_op"] else 0) ^ (1 if is_per_axis else 0),
-                )
-            )
-            new_value_infos.append(
-                onnx.helper.make_tensor_value_info(
-                    dq_output, float_elem_type, shape=[N, K] if config["use_transpose_op"] else [K, N]
-                )
+            # DequantizeLinear -> Transpose -> MatMul -> Add (optional)
+            dq = op.DequantizeLinear(
+                q_weight,
+                q_scales,
+                q_zeros,
+                block_size=None if is_per_axis else block_size,
+                # for some reason block_wise and per-axis appear to use swapped axis
+                # flip the axis if it is per-axis
+                axis=config["use_transpose_op"] or is_per_axis,
```

Comment:
still need the ^ to flip the axis. but it could be simplified to `int(config["use_transpose_op"] ^ is_per_axis)`

Reply:
We can just take a boolean and treat it as an integer, or is config["use_transpose_op"] or is_per_axis not boolean?

Reply:
if it does the cast to int automatically then int is not required. but the operator is still ^ and not or

Reply:
Oh ^ is xor sorry

Reply:
Fixed